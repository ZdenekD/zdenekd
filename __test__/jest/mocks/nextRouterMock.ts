const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        asPath: '/page/subpage/something-else',
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
