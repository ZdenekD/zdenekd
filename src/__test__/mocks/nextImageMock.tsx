const nextImageMock = jest.mock('next/image', () => function Image() {
    return <img src="" alt="" />;
});

export default nextImageMock;
