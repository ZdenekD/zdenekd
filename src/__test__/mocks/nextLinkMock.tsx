const nextLinkMock = jest.mock('next/link', () => function Link() {
    return <a href="/">link</a>;
});

export default nextLinkMock;
