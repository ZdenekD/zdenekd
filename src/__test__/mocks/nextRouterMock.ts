const nextRouterMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'en',
        locales: ['cs', 'en'],
        query: {param: 'param'},
    })),
}));

export default nextRouterMock;
