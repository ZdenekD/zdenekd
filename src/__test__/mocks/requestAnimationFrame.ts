import react from 'react';

window.requestAnimationFrame = () => 0;

export default react;
