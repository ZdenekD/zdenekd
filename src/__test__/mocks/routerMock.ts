const routerMock = jest.mock('next/router', () => ({
    useRouter: jest.fn().mockImplementation(() => ({
        pathname: '/',
        route: '/',
        locale: 'cs',
        locales: ['cs', 'en'],
    })),
}));

export default routerMock;
