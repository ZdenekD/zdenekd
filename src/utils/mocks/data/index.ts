import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
