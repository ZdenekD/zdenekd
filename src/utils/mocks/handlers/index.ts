import handler from './handler';

const handlers = [...handler];

export default handlers;
