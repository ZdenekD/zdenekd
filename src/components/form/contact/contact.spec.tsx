import {render, screen} from '@testing-library/react';
import '@/test/mocks/nextRouterMock';
import Contact from '.';

describe('Form/Contact', () => {
    it('renders without error', () => {
        render(<Contact />);

        expect(screen.getByTestId('component-form')).toBeInTheDocument();
    });
});
