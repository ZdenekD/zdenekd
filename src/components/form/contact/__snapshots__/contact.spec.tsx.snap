// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form/Contact match snapshot 1`] = `
<form class="form form"
      data-test="component-form"
>
  <div class="control   ">
    <input type="text"
           id="_34rFI1"
           name="name"
           class="input   "
           required
           maxlength="30"
           data-test="component-input"
    >
    <label for="_34rFI1"
           class="label"
    >
      <span class="labelContent">
        Jméno
        <sup class="required">
          *
        </sup>
      </span>
    </label>
    <span class="maxlength">
      0 / 30
    </span>
  </div>
  <div class="control   ">
    <input type="email"
           id="_34rFI2"
           name="email"
           class="input   "
           required
           maxlength="90"
           data-test="component-input"
    >
    <label for="_34rFI2"
           class="label"
    >
      <span class="labelContent">
        E-mail
        <sup class="required">
          *
        </sup>
      </span>
    </label>
    <span class="maxlength">
      0 / 90
    </span>
  </div>
  <div class="control   ">
    <textarea id="_34rFI3"
              name="message"
              class="textarea   "
              required
              maxlength="255"
              data-test="component-textarea"
    >
    </textarea>
    <label for="_34rFI3"
           class="label"
    >
      <span class="labelContent">
        Zpráva
        <sup class="required">
          *
        </sup>
      </span>
    </label>
    <span class="maxlength">
      0 / 255
    </span>
  </div>
  <input type="text"
         name="med"
         autocomplete="off"
         tabindex="-1"
         class="_visuallyhidden"
  >
  <button type="submit"
          class="button  submit"
          data-test="component-button"
  >
    Odeslat
  </button>
</form>
`;
